### ISSUE TITLE
feature: ability to reference modules in same github repo

### ISSUE THREAD

**Issue by billiford** [armory/dinghy#88]
I work with dinghyfiles that create several pipelines for Spinnaker. These dinghy files reach thousands of lines, making them more difficult than necessary to work with.

I understand that dinghy looks at an *external* repo for where its modules are defined, and this repo is defined in the `hal` config.

Instead of having a `module` function that references these modules in a separate github repo, would it be possible to add functionally for a `local_module` function to reference modules that are in the same repo as the dinghyfile?

Here is my current repo structure for a dinghyfile.
```
.
├── dinghyfile
└── README.md
```
Instead, this is how I would like to organize the project structure.
```
.
├── dinghyfile
├── modules/
├──── my-pipeline-name/
├────── pipeline.module
├────── stages/
├──────── my-stage-1.module
├──────── my-stage-2.module
└── README.md
```
To me, this would make the dinghyfile much easier to work with and would be a more logical directory structure.

Here's an example dinghyfile for the above directory structure.
```
{
    "application": "my-app",
    "pipelines": [
        {{ local_module "modules/my-pipeline-name/pipeline.module" }}
    ]
}
```
Here's the example `pipeline.module`.
```
{
    "application": "my-app",
    "keepWaitingPipelines": false,
    "limitConcurrent": true,
    "name": "my-pipeline-name",
    "stages": [
        {{ local_module "stages/my-stage-1.module" }},
        {{ local_module "stages/my-stage-2.module" }}
    ],
    "triggers": []
}
```

**Comment by akshayabd** [comment]
Hey @billiford, we've finished implementing this and its available in 2.19.9 release and also in ARM CLI: https://github.com/armory-io/arm/blob/master/examples/dinghyfile_localmodule

We're adding documentation and more examples, we'll let you know when thats complete.

**Comment by Jossuecito** [comment]
@billiford  you can find the full documentation for the local_module in https://docs.armory.io/docs/spinnaker/using-dinghy/#local-module-functionality
